providers/fips/fipsprov.c

```diff
@@ -235,7 +252,16 @@ static int fips_get_params(void *provctx, OSSL_PARAM params[])
     p = OSSL_PARAM_locate(params, OSSL_PROV_PARAM_BUILDINFO);
     if (p != NULL && !OSSL_PARAM_set_utf8_ptr(p, OPENSSL_FULL_VERSION_STR))
         return 0;
-
+    p = OSSL_PARAM_locate(params, OSSL_PROV_PARAM_STATUS);
+    if (p != NULL && !OSSL_PARAM_set_uint(p, FIPS_is_running()))
+        return 0;
+    p = OSSL_PARAM_locate(params, OSSL_PROV_PARAM_SELF_TEST);
+    if (p != NULL) {
+        set_self_test_cb(FIPS_get_core_handle(selftest_params.libctx));
+        status = SELF_TEST_post(&selftest_params, 1) ? 1 : 0;
```

Comment:
I find it quite strange that running the self test is a side-effect of "getting" parameters. I would have expected a separate dedicated function to trigger this.

Reply:
Ok changed.